modules/web/crawler.py

```diff
@@ -1,61 +1,79 @@
-import requests
-from modules.web.bs4 import BeautifulSoup # https://pypi.org/project/beautifulsoup4/
+from requests import get
+
+from bs4 import BeautifulSoup
+
+from modules.random_user_agent import random_user_agent
+from modules.logger import Logger
 
 
 def crawl(target_url):
-    if not target_url.endswith('/'):
-        target_url += '/'
-    urls = []
-    reqs = requests.get(target_url)
-    soup = BeautifulSoup(reqs.text, 'html.parser')
-
-    for link in soup.find_all('a'):
-        url = link.get('href')
-        if not url.startswith('http'):
-            if url.startswith('./'):
-                url = target_url + url.lstrip('./')
-            elif url.startswith('/'):
-                url = target_url + url.lstrip('/')
-            elif "#" in url:
-                continue
-            elif url == '' or url == None:
+    log = Logger()
+
+    if not target_url.endswith("/"):
+        target_url += "/"
+
+    try:
+        reqs = get(
+                target_url, headers={
+                        "User-Agent": next(random_user_agent())
+                    }
+            )
+        soup = BeautifulSoup(reqs.text, "html.parser")
+    except ConnectionError:
+        log.logger("error", "Connection error was raised, aborting.")
+        raise SystemExit
```

Comment:
We shouldn't just exit if we get a connection error.

Reply:
I set it to do 3 tries before raising systemexit since > 3 should be too much I think, although I plan to include a parameter later that would define how many trials should the program do upon meeting an exception before raising systemexit.